### ISSUE TITLE
Missing glfw3Config.cmake in Windows installer

### ISSUE THREAD

**Issue by jackjansen** [glfw/glfw#1668]
The Windows binary installer doesn't include `glfw3Config.cmake` or `glfw3-config.cmake`, therefore you can't use this installer when you want to build a cmake-based project that uses glfw.

This is rather inconvenient, because building glfw from source also pulls in its dependencies, etc.

Can I ask that a cmake config file is added to the Windows installer, so I could simply use `find_package(glfw3)`, possibly after setting the search paths so it can find the glfw3 installed directory?

**Comment by zarthcode** [comment]
Was just surprised by this as well. Workaround was to simply use as a submodule, but it would be ideal if cmake just worked.

**Comment by jackjansen** [comment]
@zarthcode do you have example code? I gave up, because the organisation of the glfw3 distribution for Windows has many different execution models/runtime systems all lumped together (each in their own subdirectory), and I couldn't work out how to use the correct CMake `if()` constructs to ensure I would be picking up the right libraries and DLLs...

**Comment by JeremyBorys** [comment]
@jackjansen Do you mean an installer generated by CPack or the windows binaries from https://www.glfw.org/download.html?

**Comment by zarthcode** [comment]
@jackjansen I've actually settled on using CMake FetchContent for cases where find_package doesn't work.  This makes configuration more robust, and keeps my git operations fairly simple as well.  It should be fairly straightforward from documentation. [FetchContent](https://cmake.org/cmake/help/git-master/module/FetchContent.html)

That said, @JeremyBorys is probably right about just using a package manager if you're looking for the fastest/easiest path to done. The conan package manager can [automatically generate](https://jfreeman.dev/blog/2019/05/22/trying-conan-with-modern-cmake:-dependencies/) cmake FMs - so that's a start.

If you're developing on windows w/vcpkg, it appears that there are steps to ["specify a toolchain file for cross-compilation](https://developerpaul123.github.io/c++/cmake/using-vcpkg-on-windows/)".  Once you've specified the vcpkg.cmake location, you should also pick up installed packages.

**Comment by jackjansen** [comment]
@JeremyBorys I meant the installers loaded from <https://www.glfw.org/download.html>. I'll update the original post.

@zarthcode and @JeremyBorys FetchContent isn't a solution for me: I don't want people that download my package to be able to build it themselves (although of course they're free to do so), I want them to be able to simply unpack my binary distribution and run it (after installing the correct glfw binary installer, either themselves or with help from an installer from me).

Basically I want people to use the same procedure as for many other packages (jpegturbo and pcl spring to mind, because I also depend on them): users just download the binary packages and install them all in the default location and when they download my binary package everything will just work together.

**Comment by JeremyBorys** [comment]
Thanks @jackjansen , I think this would be a good feature - I myself dislike zip packages. This would have to go to the guys releasing glfw because this would require changing from zip files to actual installers to place the binaries in the default location for the user like you described.

If you want a solution that will work today, a package manager (conan for example) like I suggested earlier does exactly what you are requesting. But it does add a dependency on the package manager which I am assuming you are trying to avoid.

**Comment by meshula** [comment]
I'm using FetchContent as well. Since glfw doesn't create and install glfwConfig.cmake in the usual manner, the build is slowed down a fair amount because FetchContent goes to the internet, and glfw becomes part of the dependent build (as it would be in a submodule situation), instead of a build-once part of the configuration process. This is another vote to include a config file to ease cmake builds and improve build efficiency. (I am aware of package managers, but they are not applicable to my build environment.)

**Comment by pc-john** [comment]
This glfw3Config.cmake would be very cool! Glfw is the last one among my windowing libraries (SDL3, SDL2, QT6, QT5) that does not provide glfw3Config.cmake in their Windows pre-build libraries. I workarounded it by using Find_path/Find_library/Find_file. So, glfw stays with least modern integration in my projects.
